docs/multithreading/multithreading.md

```diff
@@ -0,0 +1,43 @@
+
+- :fontawesome-solid-memory: The virtual address space is usually shared between threads. Currently, **libdebug** does not handle the threading options that separate the memory layout. The assumption is that all threads have access to the same memory space.
+
+- :material-sign-caution: Software breakpoints are implemented through code patching in the process memory. This means that a breakpoint set in one thread will affect all threads.
+    - When using [synchronous](../../stopping_events/debugging_flow) breakpoints, you will need to "diagnose" the stopping event to determine which thread triggered the breakpoint. You can do this by checking the return value of the [`hit_on()`](../../stopping_events/debugging_flow/#hit-records) method of the [Breakpoint](../../from_pydoc/generated/data/breakpoint/) object. Passing the [ThreadContext](../../from_pydoc/generated/state/thread_context/) as an argument will return `True` if the breakpoint was hit by that thread.
+
+    - When using [asynchronous](../../stopping_events/debugging_flow) breakpoints, the breakpoint will be more intuitive to handle, as the signature of the [callback function](../../stopping_events/breakpoints#callback-signature) includes the [ThreadContext](../../from_pydoc/generated/state/thread_context/) object that triggered the breakpoint.
+
+    - :octicons-cpu-24: While hardware breakpoints are thread-specific, **libdebug** mirrors them across all threads. This is done to avoid the complexity of managing hardware breakpoints on a per-thread basis. Watchpoints are hardware breakpoints, so this applies to them as well.
+
+- :fontawesome-solid-terminal: For consistency, [syscall handlers](../../stopping_events/syscalls) are also enabled across all threads. The same considerations for synchronous and asynchronous breakpoints apply here as well.
+
+!!! WARNING "Concurrency in Syscall Handling"
+    **libdebug** does not guarantee that the syscall entry and exit events will be triggered by the same thread. This is because the kernel may schedule a different thread to handle the syscall exit event.
+
+- :material-traffic-light-outline: [Signal Catching](../../stopping_events/signals) is also shared among threads. Apart from consistency, this is a necessity. In fact, the kernel does not guarantee that a signal sent to a process will be dispatched to a specific thread.
+    - By contrast, when sending arbitrary signals through the [ThreadContext](../../from_pydoc/generated/state/thread_context/) object, the signal will be sent to the requested thread.
+
+
+!!! EXAMPLE "How to access TLS?"
+    While the virtual address space is shared between threads, each thread has its own [Thread Local Storage (TLS)](https://en.wikipedia.org/wiki/Thread-local_storage) area. As it stands, **libdebug** does not provide a direct interface to the TLS area. Moreover, some of the registers that are used to access the TLS area are not exposed by **libdebug** in the `regs` attribute.
```

Comment:
We will expose `fs_base` and other registers though

Reply:
Ok, I'll change it when we have the final list of registers before the release

Reply:
[9e99adc](https://github.com/libdebug/libdebug/pull/129/commits/9e99adc7c4f740849576a6ec29d1c746c4e2a5ef)